packages/core/src/core.ts

```diff
@@ -461,7 +487,17 @@ class ThresholdKey implements ITKey {
     return tssPolyCommits;
   }
 
-  getTSSPub(): Point {
+  getTSSPub(accountIndex?: number): Point {
+    if (accountIndex && accountIndex > 0) {
+      const nonce = this.computeNonce(accountIndex);
+      // we need to add the pub key nonce to the tssPub
+      const noncePub = ecCurve.keyFromPrivate(nonce.toString("hex")).getPublic();
+      const pubKeyPoint = ecCurve
+        .keyFromPublic({ x: this.getTSSCommits()[0].x.toString("hex"), y: this.getTSSCommits()[0].y.toString("hex") })
```

Comment:
cleanup this code, calculate "this.getTSSCommits()" once and use it , rather thn calling it everytime